tink-docker/main.go

```diff
@@ -79,10 +79,31 @@ func parseCmdLine(cmdLines []string) (cfg tinkConfig) {
 // downloadFile will download a url to a local file. It's efficient because it will
 // write as it downloads and not load the whole file into memory.
 func downloadFile(filepath string, url string) error {
-	// Get the data
-	resp, err := http.Get(url)
-	if err != nil {
-		return err
+	// As all functions in the LinuxKit services run in parallel, ensure that we can fail
+	// succsefully until we accept that networking is actually broken
+
+	var retryCount int
+	var timeOut time.Duration
+	retryCount = 10
+	timeOut = time.Millisecond * 500 // 0.5 seconds
+	var resp *http.Response
+	var err error
+	// Retry this task
+	for {
```

Comment:
Normally I would call for exponential back-off w/ jitter (such as by implemented  https://github.com/cenkalti/backoff), but this seems harmless enough, given the target and limited duration.

Reply:
Rather than making this appear as a limitless loop, could you please define the bounds up front:

for retries := 0; i < 10; i++ {
  fetch ..
  if err == nil {
    break
  }
  sleep
}